### PR TITLE
chore(ci): Restrict usage of benchmark runs

### PR DESCRIPTION
**Pull request by lucperkins**
I noticed in PR #3740, which is a docs-only PR, that the benchmark suite was run. To me, the benchmarks should be run only when truly necessary. This PR adds a `paths` parameter in the benchmark Action definition so that the benchmarks are only run when the Vector sources are updated and excludes everything else, like the `README`, the docs, the `Makefile, the RFCs, etc. The selection of paths was a bit guesswork-y on my part, so please do provide feedback so we don't remove anything that needs to remain.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
Thanks @lucperkins ! I agree this shouldn't run on documentation-only PRs.

**Comment by lucperkins** [comment]
@jszwedko Thanks for the speedy review and glad you agree 😄

I'm going to wait on additional approval from @jamtur01 just to be safe.

**Comment by sghall** [comment]
Drive by 🚗 ... Would it make more sense to just ignore the `docs` path instead of listing out the other paths.  Looks like you can do that...
https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-ignoring-paths

> Anytime a path name matches a pattern in paths-ignore, the workflow will not run. GitHub evaluates patterns defined in paths-ignore against the path name. A workflow with the following path filter will only run on push events that include at least one file outside the docs directory at the root of the repository.

**Comment by lucperkins** [comment]
@sghall You're quite right about that! But in this case my intention is to restrict the benchmark runs beyond just excluding the docs.

**Comment by jszwedko** [comment]
Bump @jamtur01

**Review by jamtur01 — APPROVED** [review]
Lgtm

**Pull request merged**
